clients/js/test/extensions/interestBearingMint/getAmountToUiAmount.test.ts

```diff
@@ -0,0 +1,296 @@
+import test from 'ava';
+import type {
+  GetAccountInfoApi,
+  Lamports,
+  Rpc,
+  Base64EncodedBytes,
+  Commitment,
+  UnixTimestamp,
+} from '@solana/web3.js';
+import { address, Address } from '@solana/web3.js';
+import { getSysvarClockEncoder, SYSVAR_CLOCK_ADDRESS } from '@solana/sysvars';
+import {
+  amountToUiAmountForMintWithoutSimulation,
+  uiAmountToAmountForMintWithoutSimulation,
+  TOKEN_2022_PROGRAM_ADDRESS,
+  getMintEncoder,
+} from '../../../src';
+
+const ONE_YEAR_IN_SECONDS = 31556736;
+
+type AccountInfo = Readonly<{
+  executable: boolean;
+  lamports: Lamports;
+  owner: Address;
+  rentEpoch: bigint;
+  data: Buffer;
+}>;
+
+function getMockRpc(
+  accounts: Record<Address, AccountInfo>
+): Rpc<GetAccountInfoApi> {
+  const getAccountInfo = (
+    address: Address,
+    _config?: { commitment?: Commitment }
+  ) => {
+    const account = accounts[address];
+    if (!account) {
+      throw new Error(`Account not found for address: ${address}`);
+    }
+    if (!(account.data instanceof Uint8Array)) {
+      throw new Error(
+        `Account data is not a Uint8Array for address: ${address}`
+      );
+    }
+    return {
+      send: async () => ({
+        context: { slot: 0n },
+        value: account
+          ? {
+              executable: account.executable,
+              lamports: account.lamports,
+              owner: account.owner,
+              rentEpoch: account.rentEpoch,
+              data: [
+                Buffer.from(account.data).toString('base64'),
```

Comment:
I know these are tests but we're trying to move away from using the Buffer API. Could we change this to:

```suggestion
                getBase64Decoder().decode(account.data),
```

Same comment for anywhere we use `Buffer` in this file.